lesson 7 Homework completed

Hanna-Konovalova opened the pull request:
Homework completed

Hanna-Konovalova committed:
lesson 7 Homework completed

RomanKashenok commented on src/com/konovalova/lesson7/HomeworkTask1.java:
```diff
@@ -0,0 +1,21 @@
+         * */
+        Random rand = new Random();
+        int[] someNumbers = new int[20];
+        int i = 0;
```
давай инициализировать в цикле

RomanKashenok commented on src/com/konovalova/lesson7/HomeworkTask1.java:
```diff
@@ -0,0 +1,21 @@
+        for (; i < someNumbers.length; i++) {
+            someNumbers[i] = rand.nextInt(50);
+            System.out.print(someNumbers[i] + "  ");
+            maxNumber = Math.max(maxNumber, someNumbers[i]);
```
1 - а зачем нам на каждом заполнении вычислять?
2 - это задание было до того, как мы класс Math рвссмотрели, так что тут я жду что ты алгоритм напишешь по вычислению максимального числа, а не воспользуешься чьим-то решением

RomanKashenok commented on src/com/konovalova/lesson7/HomeworkTask2.java:
```diff
@@ -0,0 +1,64 @@
+    public static void main(String[] args) {
+        /* Создать массив оценок произвольной длины, вывести максимальную и
+         * минимальную оценку, её (их) номера.
+         * */
```
я очень плохо понимаю что ты тут делаешь по нескольким причинам: инициализация счетчиков циклов), слишком много ненужных циклом, зачем-то многомерный массив... 
Тут всё делается в 2 цикла

RomanKashenok commented on src/com/konovalova/lesson7/HomeworkTask3.java:
```diff
@@ -0,0 +1,27 @@
+        System.out.println("\nRevers array: ");
+        for(i=someNumbers.length - 1;i>=0;i--){
+            System.out.print(someNumbers[i] + "  ");
+        }
```
ты не перевернула массив, а просто распечатала его в обратном порядке

RomanKashenok commented on src/com/konovalova/lesson7/HomeworkTask5.java:
```diff
@@ -0,0 +1,32 @@
+
+            for (int j = 0; j < butterflyArray[i].length; j++) {
+                //butterflyArray[i][j] = 1;
+                butterflyArray[i][j] = validNumber(i, j);
```
а если размер массива, скажем, 11 на 11?

RomanKashenok commented on src/com/konovalova/lesson7/HomeworkTask6.java:
```diff
@@ -0,0 +1,34 @@
+
+        String[] separatedWords = ourSentence.split(" ");
+
+        HashMap<String, Integer> separatedWordsCounter = new HashMap<>();
```
мы пока не знаем ничего про HashMap

RomanKashenok commented on src/com/konovalova/lesson7/HomeworkTask7.java:
```diff
@@ -0,0 +1,37 @@
+
+                if (word != " ") {
+                    if (!separatedWordsCounter.containsKey(word)) {
+                        separatedWordsCounter.put(word, 0);
```
почему с нулем?

RomanKashenok commented on src/com/konovalova/lesson7/HomeworkTask6.java:
```diff
@@ -0,0 +1,34 @@
+        for (String word : separatedWords)
+        {
+            if (!separatedWordsCounter.containsKey(word)){
+                separatedWordsCounter.put(word, 0);
```
почему с нулем?

RomanKashenok commented on src/com/konovalova/lesson7/HomeworkTask8.java:
```diff
@@ -0,0 +1,130 @@
+            if((playGround[x][0]==playerMark & playGround[x][1]==playerMark & playGround[x][2]==playerMark) ||
+               (playGround[0][x]==playerMark & playGround[1][x]==playerMark & playGround[2][x]==playerMark))
+            {
+                winResult = true;
```
тут я думаю можно сразу return true ставить. Да и в целом избавиться от переменной winResult

RomanKashenok commented on src/com/konovalova/lesson7/HomeworkTask8.java:
```diff
@@ -0,0 +1,130 @@
+
+        printPlayGroun(playGround);
+
+        while (true) {
```
у тебя одинаковый флоу проверки для человека и для бота. Вынеси обе проверки в отдельный метод и вызывай его просто с аргументом 2 или 3 соответственно - не придется дублировать код

Hanna-Konovalova replied:
Что такое "флоу проверки"? Для человека и для бота создан один метод "isWin", он работает и для человека и для бота, в зависимости от аргумента, который передается. Это имеется в виду?

RomanKashenok replied:
я про то, что вот это можно не дублировать, а вынести в отдельный метод:
            if(isWin(2,playGround)){
                System.out.println("You win");
                printPlayGroun(playGround);
                break;
            } else {
                //печать поля
                printPlayGroun(playGround);
            }
            //проверка ничья
            if (noWins(playGround)){
                System.out.println("No wins");
                break;
            }

дв адубликата этих строк отличаются только цифрой 2 и 3 соответственно

Hanna-Konovalova committed:
lesson 7 HomeworkTask7 - add variable initialization in the loop add change algorithm for max number search
lesson 7 HomeworkTask2 change algorithm for max and min number search
lesson 7 HomeworkTask3 add creation of inverted array
lesson 7 HomeworkTask5 make universal algorithm for butterfly array
lesson 7 HomeworkTask6 change algorithm for frequency dictionary without using HashMap
lesson 7 HomeworkTask7 change algorithm for frequency dictionary without using HashMap
lesson 7 HomeworkTask8 delete boolean winResult

Hanna-Konovalova committed:
lesson 7 HomeworkTask7 - add variable initialization in the loop add change algorithm for max number search
lesson 7 HomeworkTask2 change algorithm for max and min number search
lesson 7 HomeworkTask3 add creation of inverted array
lesson 7 HomeworkTask5 make universal algorithm for butterfly array
lesson 7 HomeworkTask6 change algorithm for frequency dictionary without using HashMap
lesson 7 HomeworkTask7 change algorithm for frequency dictionary without using HashMap
lesson 7 HomeworkTask8 delete boolean winResult

RomanKashenok commented on src/com/konovalova/lesson7/HomeworkTask6.java:
```diff
@@ -0,0 +1,55 @@
+
+        //1. Заполнение массива будущих уникальных значений пустыми данными
+        for(int i=0;i<uniqWords.length;i++){
+            uniqWords[i] = "";
```
вот этого не надо делать

RomanKashenok commented on src/com/konovalova/lesson7/HomeworkTask6.java:
```diff
@@ -0,0 +1,55 @@
+            uniW = separatedWords[i];
+            isUniq = true;
+            for (int j = 0; j < uniqWords.length; j++) {
+                if (uniqWords[j].equals(uniW)) {
```
в незаполненном массиве надо будет проверить, что uniqWords[j] != null

RomanKashenok commented on src/com/konovalova/lesson7/HomeworkTask7.java:
```diff
@@ -0,0 +1,60 @@
+
+            //1. Заполнение массива будущих уникальных значений пустыми данными
+            for(int i=0;i<uniqWords.length;i++){
+                uniqWords[i] = "";
```
ну как и в предыдущем

RomanKashenok commented on src/com/konovalova/lesson7/HomeworkTask8.java:
```diff
@@ -0,0 +1,128 @@
+        for(int x = 0;x<3;x++){
+            for(int y = 0;y<3;y++){
+                if(playGround[x][y]==0){
+                    noWinResult = false;
```
return false; (break не нужен)

RomanKashenok commented on src/com/konovalova/lesson7/HomeworkTask8.java:
```diff
@@ -0,0 +1,128 @@
+                }
+            }
+        }
+        return noWinResult;
```
return true;

Hanna-Konovalova committed:
lesson7 HomeworkTask6,7 delete initial filling of an empty array
lesson7 HomeworkTask8 add common method for user/computer acting
